src/test/java/com/algolia/search/apache/analytics/AnalyticsTest.java

```diff
@@ -0,0 +1,26 @@
+package com.algolia.search.apache.analytics;
+
+import com.algolia.search.AnalyticsClient;
+import com.algolia.search.integration.IntegrationTestExtension;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.IOException;
+
+import static com.algolia.search.integration.IntegrationTestExtension.ALGOLIA_API_KEY_1;
+import static com.algolia.search.integration.IntegrationTestExtension.ALGOLIA_APPLICATION_ID_1;
+
+@ExtendWith({IntegrationTestExtension.class})
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class AnalyticsTest extends com.algolia.search.integration.analytics.AnalyticsTest {
+
+  AnalyticsTest() {
+    super(IntegrationTestExtension.searchClient, new AnalyticsClient(ALGOLIA_APPLICATION_ID_1, ALGOLIA_API_KEY_1));
```

Comment:
Here `IntegrationTestExtension.searchClient`is `null` that's why tests are failing. If I remove `@TestInstance(TestInstance.Lifecycle.PER_CLASS)` and move `afterAll()` in the implementation I don't have the issue anymore.